src/sdk/pynni/tests/test_compressor.py

```diff
@@ -130,22 +131,24 @@ def test_torch_fpgm_pruner(self):
         """
 
         model = TorchModel()
-        config_list = [{'sparsity': 0.2, 'op_types': ['Conv2d']}, {'sparsity': 0.6, 'op_types': ['Conv2d']}]
-        pruner = torch_compressor.FPGMPruner(model, config_list)
+        optimizer = torch.optim.SGD(model.parameters(), lr=0.01, momentum=0.5)
+        config_list = [{'sparsity': 0.2, 'op_types': ['Conv2d']}]
+        pruner = torch_compressor.FPGMPruner(model, config_list, optimizer)
 
-        model.conv2.weight.data = torch.tensor(w).float()
-        layer = torch_compressor.compressor.LayerInfo('conv2', model.conv2)
-        masks = pruner.calc_mask(layer, config_list[0], if_calculated=torch.tensor(0))
-        assert all(torch.sum(masks['weight'], (1, 2, 3)).numpy() == np.array([45., 45., 45., 45., 0., 0., 45., 45., 45., 45.]))
+        model.conv2.module.weight.data = torch.tensor(w).float()
+        masks = pruner.calc_mask(model.conv2)
+        assert all(torch.sum(masks['weight_mask'], (1, 2, 3)).numpy() == np.array([45., 45., 45., 45., 0., 0., 45., 45., 45., 45.]))
 
-        model.conv2.weight.data = torch.tensor(w).float()
-        masks = pruner.calc_mask(layer, config_list[1], if_calculated=torch.tensor(0))
-        assert all(torch.sum(masks['weight'], (1, 2, 3)).numpy() == np.array([45., 45., 0., 0., 0., 0., 0., 0., 45., 45.]))
+        model.conv2.module.weight.data = torch.tensor(w).float()
+        model.conv2.if_calculated = False
+        model.conv2.config = {'sparsity': 0.6, 'op_types': ['Conv2d']}
+        masks = pruner.calc_mask(model.conv2)
+        assert all(torch.sum(masks['weight_mask'], (1, 2, 3)).numpy() == np.array([45., 45., 0., 0., 0., 0., 0., 0., 45., 45.]))
 
     @tf2
     def test_tf_fpgm_pruner(self):
         model = get_tf_model()
-        config_list = [{'sparsity': 0.2, 'op_types': ['Conv2D']}, {'sparsity': 0.6, 'op_types': ['Conv2D']}]
```

Comment:
cannot remove the second dict, it is testing which `sparsity` takes effect

Reply:
I think this test should be moved to additional test for config selection test.

Reply:
good point. could you add this one to the config selection test? and we can add more unit tests to it in future.

Reply:
Of course.